processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -218,35 +215,16 @@ public PropertyMappingBuilder forgedNamedBased(boolean forgedNamedBased) {
             return this;
         }
 
-        public PropertyMappingBuilder nullValueCheckStrategy(NullValueCheckStrategyPrism nvcs ) {
-            this.nvcs = nvcs;
-            return this;
-        }
-
-        public PropertyMappingBuilder nullValuePropertyMappingStrategy( NullValuePropertyMappingStrategyPrism nvpms ) {
-            this.nvpms = nvpms;
+        public PropertyMappingBuilder options(DelegatingOptions options) {
+            this.nvcs = options.getNullValueCheckStrategy();
+            if ( method.isUpdateMethod() ) {
```

Comment:
Why not store the `DelegatingOptions`? Right now this forces us to pass `method` before passing the options and can lead to an NPE.

Reply:
I guess there's room for improvement (luckily). For instance, we only use the `NullValuePropertyMapping` when its an update method. There's another issue, that will align this in 2.0.  This PR was big enough on its own right. Hence I left this out.